diff --git a/node_modules/@bcherny/json-schema-ref-parser/cjs/util/url.js b/node_modules/@bcherny/json-schema-ref-parser/cjs/util/url.js
index b7b0cbd..4ed15ca 100644
--- a/node_modules/@bcherny/json-schema-ref-parser/cjs/util/url.js
+++ b/node_modules/@bcherny/json-schema-ref-parser/cjs/util/url.js
@@ -155,17 +155,18 @@ function isFileSystemPath(path) {
     var protocol = getProtocol(path);
     return protocol === undefined || protocol === "file";
 }
+const { join, isAbsolute } = require('path')
 function fromFileSystemPath(path) {
     // Step 1: On Windows, replace backslashes with forward slashes,
     // rather than encoding them as "%5C"
     if (isWindows) {
         var hasProjectDir = path.toUpperCase().includes(_projectDirCjs.default.replace(/\\/g, "\\").toUpperCase());
         var hasProjectUri = path.toUpperCase().includes(_projectDirCjs.default.replace(/\\/g, "/").toUpperCase());
-        if (hasProjectDir || hasProjectUri) {
-            path = path.replace(/\\/g, "/");
-        } else {
-            path = "".concat(_projectDirCjs.default, "/").concat(path).replace(/\\/g, "/");
+        var isAbsolutePath = isAbsolute(path)
+        if(!(hasProjectUri || hasProjectDir || isAbsolutePath)) {
+           path = join(_projectDirCjs.default, path)
         }
+        path = path.replace(/\\/g, "/");
     }
     // Step 2: `encodeURI` will take care of MOST characters
     path = encodeURI(path);
